docs/proposals/20-ha-control-planes.md

```diff
@@ -0,0 +1,1259 @@
+      - [Recovery from Zone failure](#recovery-from-zone-failure)
+    - [Option #1: Leverage existing recovery options - `Preferred`](#option-1-leverage-existing-recovery-options---preferred)
+    - [Option #2: Redundencies for all critical control plane components](#option-2-redundencies-for-all-critical-control-plane-components)
+    - [Option #3: Auto-rebalance pods in the event of AZ failure](#option-3-auto-rebalance-pods-in-the-event-of-az-failure)
+  - [Cost Implications on hosting HA control plane](#cost-implications-on-hosting-ha-control-plane)
+    - [Compute & Storage](#compute--storage)
+    - [Network latency](#network-latency)
+    - [Cross-Zonal traffic](#cross-zonal-traffic)
+      - [Ingress/Egress traffic analysis](#ingressegress-traffic-analysis)
+      - [Optimizing Cost: Topology Aware Hint](#optimizing-cost-topology-aware-hint)
+  - [References](#references)
+  - [Appendix](#appendix)
+    - [ETCD Active-Passive Options](#etcd-active-passive-options)
+    - [Topology Spread Constraints evaluation and findings](#topology-spread-constraints-evaluation-and-findings)
+    - [Availability Zone Outage simulation](#availability-zone-outage-simulation)
+    - [Ingress/Egress Traffic Analysis Details](#ingressegress-traffic-analysis-details)
+
+## Summary
+
+Gardener today only offers highly available control planes for some of its components (like Kubernetes API Server and Gardener Resource Manager) which are deployed with multiple replicas which allow the spread of the replicas across nodes. Many of the other critical control plane components including `etcd` are only offered with a single replica, making them susceptible to both node failure as well as zone failure causing downtimes.
+
+This GEP extends the failure domain tolerance for shoot control plane components as well as seed components to survive extensive node or availability zone (AZ) outages.
+
+## Motivation
+
+High availability (HA) of Kubernetes control planes is desired to ensure continued operation, even in the case of partial failures of nodes or availability zones. Tolerance to common failure domains ranges from hardware (e.g. utility power sources and backup power sources, network switches, disk/data, racks, cooling systems etc.) to software.
+
+Each consumer therefore needs to decide on the degree of failure isolation that is desired for the control plane of their respective shoot clusters.
+
+## Goals
+
+- Provision shoot clusters with high available control planes (HA shoots) with failure tolerance on node or AZ level. Consumers may enable/disable high-availability and choose failure tolerance between multiple nodes within a single zone or multiple nodes spread across multiple zones.
+- Migrating existing shoots to HA shoots.
+- Scheduling HA shoots to adequate seeds.
+- Detect and mitigate AZ outages.
+
+## Non-Goals
+
+- Setting up a high available Gardener service.
+- Upgrading from a single-zone shoot control plane to a multi-zonal shoot control plane.
```

Comment:
What's the difference to above "Migrating existing shoots to HA shoots". The language seems different in any case.

Reply:
It has been reworded. Please check again